### ISSUE TITLE
RichTextInput fails to render

### ISSUE THREAD

**Issue by waynebloss** [marmelab/react-admin#2631]
**What you were expecting:**
I installed `ra-input-rich-text`, imported it into my resource file and placed it in a `SimpleForm`. I expected it to render.

**What happened instead:**
Got an error:

```js
RichTextInput.prototype.render = function () {
    var _a = this.props.meta,
        error = _a.error, // <-- Uncaught TypeError: Cannot read property 'error' of undefined
```

**Steps to reproduce:**
Install `ra-input-rich-text`, import it into a resource file and place it in a `SimpleForm`.

**Related code:**

```jsx
export const ThingEdit = props => (
  <Edit {...props}>
    <SimpleForm>
      <DisabledInput source="id" />
      <TextInput source="name" />
      <RichTextInput source="body" />
    </SimpleForm>
  </Edit>
);
```

**Environment**

* React-admin version: 2.5.1
* Last version that did not exhibit the issue (if applicable): none
* React version: 16.6.3
* Browser: Chrome 70
* Stack trace (in case of a JS error):

```
Uncaught TypeError: Cannot read property 'error' of undefined
    at RichTextInput.render (index.js:80)
    at finishClassComponent (react-dom.development.js:14899)
    at updateClassComponent (react-dom.development.js:14863)
    at beginWork (react-dom.development.js:15716)
    at performUnitOfWork (react-dom.development.js:18750)
    at workLoop (react-dom.development.js:18791)
    at HTMLUnknownElement.callCallback (react-dom.development.js:147)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:196)
    at invokeGuardedCallback (react-dom.development.js:250)
    at replayUnitOfWork (react-dom.development.js:17998)
    at renderRoot (react-dom.development.js:18909)
    at performWorkOnRoot (react-dom.development.js:19812)
    at performWork (react-dom.development.js:19722)
    at performSyncWork (react-dom.development.js:19696)
    at requestWork (react-dom.development.js:19551)
    at scheduleWork (react-dom.development.js:19358)
    at Object.enqueueSetState (react-dom.development.js:12789)
    at Connect.push../node_modules/react/cjs/react.development.js.Component.setState (react.development.js:354)
    at Connect.onStateChange (connectAdvanced.js:245)
    at Object.notify (Subscription.js:29)
    at Subscription.notifyNestedSubs (Subscription.js:66)
    at Connect.onStateChange (connectAdvanced.js:242)
    at Object.notify (Subscription.js:29)
    at Subscription.notifyNestedSubs (Subscription.js:66)
    at Connect.onStateChange (connectAdvanced.js:242)
    at Object.notify (Subscription.js:29)
    at Subscription.notifyNestedSubs (Subscription.js:66)
    at Connect.onStateChange (connectAdvanced.js:242)
    at Object.notify (Subscription.js:29)
    at Subscription.notifyNestedSubs (Subscription.js:66)
    at Connect.onStateChange (connectAdvanced.js:242)
    at Object.dispatch (createStore.js:175)
    at dispatch (<anonymous>:1:40228)
    at middleware.js:13
    at formMiddleware.js:22
    at middleware.js:72
    at dispatch (applyMiddleware.js:46)
    at utils.js:282
    at proc.js:557
    at exec (scheduler.js:20)
    at flush (scheduler.js:63)
    at asap (scheduler.js:35)
    at runPutEffect (proc.js:553)
    at runEffect (proc.js:496)
    at next (proc.js:366)
    at currCb (proc.js:449)
```

**Comment by kindleton** [comment]
I'm facing the exact same issue.
I searched on SO for this problem but could not find a solution.
I also went through the Docs repeatedly, to see if I am missing something.
I found this in the docs : https://marmelab.com/react-admin/CreateEdit.html#the-create-and-edit-components 
As you can see in the sample code, the forms have included the RichTextInput component but it is not shown rendering a WYSIWIG editor in the reference image underneath.

I would really appreciate some help with this issue. I am running on a deadline. :(

**Comment by waynebloss** [comment]
@kindleton Here's my TinyMCE setup, to save you some time:

1. `npm install @tinymce/tinymce-react` or `yarn add @tinymce/tinymce-react`
2. If you're using Create-React-App, put in your `public/index.html` file, in the `<head>` section somewhere:
`<script src="//cdn.tinymce.com/4/tinymce.min.js"></script>`
3. Render the editor in your react-admin input implementation:
```jsx
import { Editor } from '@tinymce/tinymce-react';
<Editor
  disabled={isContentAvailable}
  init={{
    // The auto_focus property doesn't actually work at this time.
    // See hackToDisableAutoFocusOnTinyMCE comments.
    auto_focus: false,
    height: 512,
    content_style: "a { color: #53B7E8; text-decoration: none }",
    plugins: "autolink code colorpicker fullscreen " + "image media link lists paste " + "preview table textcolor",
    menubar: "view edit insert format tools table", // See https://www.tiny.cloud/docs/advanced/editor-control-identifiers/
    toolbar: "undo redo | " +
      "styleselect | " +
      "bold italic | " +
      "fontselect | fontsizeselect | " +
      "alignleft aligncenter alignright alignjustify | " + "bullist numlist outdent indent | " + "link image | " + "forecolor backcolor",
    images_upload_handler: this.onEditorImageUpload,
    plugin_preview_height: 650, // default: 500,
    plugin_preview_width: 845, // default: 650,
    style_formats: [
      { title: "Subtitle", block: "h3" },
      {
        title: "Button",
        inline: "span",
        styles: {
          "background-color": "#53B7E8",
          "border-radius": "5px",
          color: "white",
          display: "inline-block",
          padding: "12px 8px",
        },
      },
    ],
  }}
  value={content}
  onEditorChange={actions.changeContent}
/>
```
4. If the `auto_focus` setting doesn't work for you, try my `hackToDisableAutoFocusOnTinyMCE` [function](https://github.com/tinymce/tinymce-react/issues/41#issuecomment-442199998) which you can call on `componentDidMount`.

**Comment by waynebloss** [comment]
@kindleton If you need image uploads, you can simply do them with Axios. Here's my code to integrate between my own uploader logic and TinyMCE:

```js
onEditorImageUpload = async (blobInfo, success, failure) => {
  // Get the `File` web API interface, just like any file input provides...
  const file = blobInfo.blob();
  // The next function just uses Axios to upload to Azure/GCS/S3/etc. and then
  // it returns the URL of where the file can be displayed by the editor after
  // upload...
  const fileURL = await uploadImageWithAxiosAndGetURL(file).catch(err => {
    failure(err);
    return undefined;
  });
  if (fileURL) {
    success(fileURL);
  }
};
```

**Comment by kindleton** [comment]
@waynebloss  Thanks a ton for all your inputs. After posting the comment, I  tried using external Input Component based on TinyMCE [aor-tinymce-input](https://github.com/LoicMahieu/aor-tinymce-input) ,which is pointed out in the react-admin docs. Apparently, that gave some errors too. I was about to sleep when I got notified of your comments and came online to try them out. I'll get back to you, over email, after trying your steps. Let's keep this issue thread open for the repo contributors to notice. :)
Thanks again! :D

**Comment by waynebloss** [comment]
@kindleton OK, so I looked at that one earlier and I didn't plan on using it because it depends on `react-tinymce` instead of `@tinymce/tinymce-react` (the latter of which is newer and built by the team that maintains the core TinyMCE package).

Before I used the `@tinymce` package, I tried to use `react-tinymce` because it was the first one that I found. However, it had some severe problems when I used it to swap the `value={content}` variable depending on what "section" of content my user selected to edit...

I would strongly recommend that you take that `aor-tinymce-input` package and convert it to use the `@tinymce/tinymce-react`.

**Comment by fzaninotto** [comment]
I can't reproduce your bug on CodeSandbox, see https://codesandbox.io/s/61m6lyvjwn. It uses the same versions you are mentioning. When editing a post, the second tab "body" shows a working rich text input.

Please fork this CodeSandbox to show a failing use case, otherwise it means the bug is on your side.

**Comment by waynebloss** [comment]
I found the problem! Thanks @fzaninotto.

Comparing my code with yours, I can see that you did ` import RichTextInput from 'ra-input-rich-text'` while I did `import { RichTextInput } from 'ra-input-rich-text'`. Looking at [the code](https://github.com/marmelab/react-admin/blob/master/packages/ra-input-rich-text/src/index.js) I can see that I was using a valid export, just the wrong one.

If I had followed [the docs](https://marmelab.com/react-admin/Inputs.html#richtextinput) to the letter, I wouldn't have had this problem. However I can't blame myself too much since the code compiled and had no warnings or errors and it seemed to follow the pattern of the rest of `react-admin` where we import things with brackets. I didn't even think to try the default export (but I also wasn't trying very hard to get this working - just exploring mostly).
